### ISSUE TITLE
Use composer.json in favor of module.json if config flag is set.

### ISSUE THREAD

**Issue by IngwiePhoenix** [caffeinated/modules#110]
Laravel internally already uses this structure - using composer inside through the outside, to do dependency management and package idendification. I do see the point of using `module.json`, but wouldn't it make more sense if one could also just use a `composer.json` file instead?

i.e.:
```php
<? return [
    "manifestFile"=>"composer.json",
    "slugKey"=>"name"
];
```

How about it?

**Comment by kaidesu** [comment]
I don't think it makes more sense - the `composer.json` file is to manage and set dependencies and information specifically for Composer-based packages. While you _may_ add a `composer.json` to your modules (see [here](https://github.com/caffeinated/modules/wiki/Composer)) - these modules are not composer packages.

The `module.json` manifest files are for anything that may be specific for your given application. Some developers keep menu structures and other information for their modules in there for example.

**Comment by IngwiePhoenix** [comment]
Well, to me it felt like a duplication. For instance, I have coded my app in a way that I have even opensourced internals and turned them into public modules. I would like to use one file for version, name, description and other info. As the composer spec reads, one can use the "extra" field. For instance, this JSON is appearantly valid:

```json
{
    "name": "DragonsInn/foo-package",
    "version": "0.1.0",
    "description": "A demo",
    "extra": {
        "slug": "foo"
    }
}
```

But in fact, you don't need to use the `extra` key at all. Just adding `"foo":"bar"` to the root level worked just fine in a test I did. However, validating the JSON fails this way. Putting it in `extras` does not.

That is why I might end up extending this in my local project to obtain this feature. I just thought I'd bring it up here since it came to my mind. :)
